x-pack/test/api_integration/apis/infra/log_entry_highlights.ts

```diff
@@ -185,236 +172,5 @@ export default function({ getService }: FtrProviderContext) {
-          expect(logEntryHighlightSet.entries).to.have.length(50);
-          expect(isSorted(ascendingTimeKey)(logEntryHighlightSet.entries)).to.equal(true);
-
-          for (const logEntryHighlight of logEntryHighlightSet.entries) {
-            expect(logEntryHighlight.columns).to.have.length(3);
-            expect(logEntryHighlight.columns[1]).to.have.property('field');
-            expect(logEntryHighlight.columns[1]).to.have.property('highlights');
-            expect(logEntryHighlight.columns[1].highlights).to.eql([
-              'generate_test_data/simple_logs',
-            ]);
-            expect(logEntryHighlight.columns[2]).to.have.property('message');
-            expect(logEntryHighlight.columns[2].message).to.be.an('array');
-            expect(logEntryHighlight.columns[2].message.length).to.be(1);
-            expect(logEntryHighlight.columns[2].message[0].highlights).to.eql([]);
-          }
-        });
-
-        it('should apply the filter query in addition to the highlight query', async () => {
-          const {
-            data: {
-              source: { logEntryHighlights },
-            },
-          } = await client.query<any>({
-            query: logEntryHighlightsQuery,
-            variables: {
-              sourceId: 'default',
-              startKey: KEY_BEFORE_START,
-              endKey: KEY_AFTER_END,
-              filterQuery: JSON.stringify({
-                multi_match: { query: 'host-a', type: 'phrase', lenient: true },
-              }),
-              highlights: [
-                {
-                  query: 'message',
-                  countBefore: 0,
-                  countAfter: 0,
-                },
-              ],
-            },
-          });
-
-          expect(logEntryHighlights).to.have.length(1);
-
-          const [logEntryHighlightSet] = logEntryHighlights;
-          expect(logEntryHighlightSet).to.have.property('entries');
-          // half of the documenst
-          expect(logEntryHighlightSet.entries).to.have.length(25);
-          expect(isSorted(ascendingTimeKey)(logEntryHighlightSet.entries)).to.equal(true);
-
-          for (const logEntryHighlight of logEntryHighlightSet.entries) {
-            expect(logEntryHighlight.columns).to.have.length(3);
-            expect(logEntryHighlight.columns[1]).to.have.property('field');
-            expect(logEntryHighlight.columns[1]).to.have.property('highlights');
-            expect(logEntryHighlight.columns[1].highlights).to.eql([]);
-            expect(logEntryHighlight.columns[2]).to.have.property('message');
-            expect(logEntryHighlight.columns[2].message).to.be.an('array');
-            expect(logEntryHighlight.columns[2].message.length).to.be(1);
-            expect(logEntryHighlight.columns[2].message[0].highlights).to.eql([
-              'message',
-              'message',
-            ]);
-          }
-        });
-
-        it('should return highlights outside of the interval when requested', async () => {
```

Comment:
It isn't. The new API is meant to be used with the dates selected by the user so I don't think this functionality is needed anymore.

Reply:
How do we fetch the highlights beyond the first and last loaded log entry with the new API? This is needed for the prev/next buttons for the highlights to be able to jump long distances to targets that haven't been loaded as part of the normal pagination.

Reply:
Mmm, true.

I don't wanna return entries that fall outside the datepicker range, because that opens a number of interesting edge cases. If there are highlights outside the datepicker range and the user clicks "Up" should we extend the range? Up to what point? What if the previous entry was a month ago? Etc.

It still should return entries outside the visible range (entries between `topCursor` and `bottomCursor`) so jumping still works.

Given `startDateExpression`, `endDateExpression`, and `entries` as the array of visible entries, we could do something like this

```ts
const middleEntry = entries[Math.floor(entries.length / 2)];

fetchLogEntriesHighlights({
  startTimestamp: datemathToEpoch(startDateExpression),
  endTimestamp: datemathToEpoch(endDateExpression),
  center: middleEntry.cursor,
  size: entries.length + 2,
})
```

This ensures that:
- No highlights will ever be loaded outside the datepicker range.
- The `size: entries.length + 2` ensures we will always return at least one highlighted entry outside the visible range, even in the edge case where all visible entries contain the highlighted term.

One potential downside is that if the visible range is long (i.e. 1000+ entries) we might have a very long request. We could minimise this by limiting how many entries are cached inside `logEntriesState`.

Do you see any downsides?

Reply:
Yes, we shouldn't fetch outside of the selected date range.

As an alternative to the large request we could issue two very small highlights requests with before and after with a size of 1 respectively. That would only fetch exactly what we need and makes fewer assumptions about the `middleEntry` and `size`.